src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -374,11 +404,18 @@ private String generateFilesJSON(List<StagedFileWrapper> files) {
   }
 
   /**
-   * addUserAgent - adds the user agent header to a request
+   * Add user agent to the request Header for passed Http request
    *
    * @param request the URI request
+   * @param customUserAgent adds the user agent header to a request along with the default one. If
+   *     it is null or empty, only default one is used.
    */
-  private static void addUserAgent(HttpUriRequest request) {
+  private static void addUserAgent(HttpUriRequest request, String customUserAgent) {
+    if (!Strings.isNullOrEmpty(customUserAgent)) {
+      final String userAgent = buildCustomUserAgent(customUserAgent);
```

Comment:
Another advantage of passing it via the constructor - we can prevent constructing this every time

Reply:
Umm, that would work as well, but I thought it would be good to have it  request level granularity instead of SimpleIngestManager ctor level. Thoughts?